pkg/api/aim/query/clause.go

```diff
@@ -52,3 +56,146 @@ func (regexp Regexp) writeColumn(builder clause.Builder) {
 		builder.WriteQuoted(regexp.Column)
 	}
 }
+
+// Json clause for string match at a json path.
+type Json struct {
+	clause.Column
+	JsonPath  string
+	Dialector string
+}
+
+// Build builds positive statement.
+func (json Json) Build(builder clause.Builder) {
+	json.writeColumn(builder)
+	jsonPath := json.JsonPath
+	switch json.Dialector {
+	case postgres.Dialector{}.Name():
+		//nolint:errcheck,gosec
+		builder.WriteString("#>>")
+		jsonPath = "{" + strings.ReplaceAll(jsonPath, ",", ".") + "}"
+	default:
+		//nolint:errcheck,gosec
+		builder.WriteString("->>")
+	}
+	builder.AddVar(builder, jsonPath)
+}
+
+// NegationBuild builds negative statement.
+func (json Json) NegationBuild(builder clause.Builder) {
+	json.writeColumn(builder)
+	jsonPath := json.JsonPath
+	switch json.Dialector {
+	case postgres.Dialector{}.Name():
+		//nolint:errcheck,gosec
+		builder.WriteString("#>>")
+		jsonPath = "{" + strings.ReplaceAll(jsonPath, ",", ".") + "}"
+	default:
+		//nolint:errcheck,gosec
+		builder.WriteString("->>")
+	}
+	builder.AddVar(builder, jsonPath)
+}
+
+func (json Json) writeColumn(builder clause.Builder) {
+	switch json.Dialector {
+	case sqlite.Dialector{}.Name():
+		//nolint:errcheck,gosec
+		builder.WriteString("IFNULL(")
+		builder.WriteQuoted(json.Column)
+		//nolint:errcheck,gosec
+		builder.WriteString(", JSON('{}'))")
+	default:
+		builder.WriteQuoted(json.Column)
+	}
+}
+
+type JsonEq struct {
+	Left  Json
+	Value any
+}
+
+func (eq JsonEq) Build(builder clause.Builder) {
+	eq.Left.Build(builder)
```

Comment:
i had to "override" these methods from the Gorm built-in so we could call `Build` on the left-hand portion, instead of writing quoted